jenkins/test-history/gen_html.py

```diff
@@ -14,340 +14,218 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Creates an HTML report for all jobs starting with a given prefix.
+"""Creates an HTML report out of the JSON."""
 
-Reads the JSON from tests.json, and prints the HTML to stdout.
-
-This code is pretty nasty, but gets the job done.
-
-It would be really spiffy if this used an HTML template system, but for now
-we're old-fashioned. We could also generate these with JS, directly from the
-JSON. That would allow custom filtering and stuff like that.
-"""
 
+from __future__ import division
 from __future__ import print_function
 
+
 import argparse
-import cgi
 import collections
+import jinja2
 import json
 import os
-import re
-import string
 import sys
 import time
 
 
-TestMetadata = collections.namedtuple('TestMetadata', [
-    'okay',
-    'unstable',
-    'failed',
-    'skipped',
-])
-
-
-PREFIX_TO_URL = {}
-
-def prefix_for(prefix):
-    if 'azure' in prefix:
-        new = 'azure$'
-    elif 'rktnetes' in prefix:
-        new = 'rktnetes$'
-    elif 'gs://kubernetes-jenkins/' in prefix:
-        new = ''
-    else:
-        new = prefix[5:-1].replace('/', '_') + '$'
-    PREFIX_TO_URL[new] = prefix
-    return new
+BLOCKING_JOBS = [
+    'kubelet-gce-e2e-ci',
+    'kubernetes-build',
+    'kubernetes-e2e-gce',
+    'kubernetes-e2e-gce-scalability',
+    'kubernetes-e2e-gce-slow',
+    'kubernetes-e2e-gke',
+    'kubernetes-e2e-gke-slow',
+    'kubernetes-kubemark-5-gce',
+    'kubernetes-kubemark-500-gce',
+    'kubernetes-test-go',
+]
 
 
-def slugify(inp):
-    """
-    Convert a string into a url-safe slug fragment.
-
-    This matches the slugify code in Gubernator.
-    """
-    inp = re.sub(r'[^\w\s-]+', '', inp)
-    return re.sub(r'\s+', '-', inp).lower()
+JobSummary = collections.namedtuple('JobSummary', [
+    'name',
+    'passed',
+    'failed',
+    'latest_failure',
+    'tests',
+    'stable',
+    'unstable',
+    'broken',
+])
 
 
-def gubernator_url(suite, build_number, test_name):
+def gubernator_url(bucket, job, build):
```

Comment:
can you retain the test_name link? on builds with many different failures, it saves time.